packages/flutter/lib/src/material/tabs.dart

```diff
@@ -2008,18 +2008,29 @@ class _TabBarViewState extends State<TabBarView> {
     _updateChildren();
   }
 
+  // Checks we can call methods like jumpToPage - which requires we have a
+  // position and viewport dimension.
+  bool _pageCanChange() {
+    return _pageController != null
+      && _pageController!.hasClients
+      && _pageController!.position.hasViewportDimension;
+  }
+
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
     _updateTabController();
     _currentIndex = _controller!.index;
-    if (_pageController == null) {
+    if (_pageCanChange()) {
+      _pageController!.jumpToPage(_currentIndex!);
+    } else {
+      // Dispose if we had one already, this can happen if the index of the tab
+      // controller, which sets _currentIndex, changes before we have laid out.
+      _pageController?.dispose();
       _pageController = PageController(
```

Comment:
Instead of doing this, can _pageController somehow sets its _PagePosition's _pageToUseOnStartup if the viewport dimension is not yet applied?